README.md

```diff
@@ -2142,6 +2264,42 @@ authClient.tokenManager.off('renewed');
 authClient.tokenManager.off('renewed', myRenewedCallback);
 ```
 
+### `authStateManager`
+
+`AuthStateManager` evaluates and emits `AuthState` based on the events from `TokenManager` for downstream clients to consume.
+
+The emitted `AuthState` object includes:
+
+* `isPending`: true in the time after page load (first render) but before the asynchronous methods to see if the tokenManager is aware of a current authentication.
+* `isAuthenticated`: true if the user is considered authenticated. Normally this is true if both an idToken and an accessToken are present in the tokenManager, but this behavior can be overridden if you passed an isAuthenticated callback in the [configuration](#configuration-reference).
+* `accessToken`: the JWT accessToken for the currently authenticated user (if provided by the scopes).
+* `idToken`: the JWT idToken for the currently authenticated user (if provided by the scopes).
+* `error`: contains the error returned if an error occurs in the `authState` evaluation process.
+
+Subscribes to `authStateChange` event:
+
+```javascript
+authClient.authStateManager.subscribe((authState) => {
+  // handle the latest evaluated authState, like integrate with client framework's state management store
+});
+```
+
+#### `authStateManager.getAuthState()`
+
+Gets latest evaluated `authState` from the `authStateManager`. The `authState` (a unique new object) is re-evaluated when `authStateManager.updateAuthState()` is called.
+
+#### `authStateManager.updateAuthState()`
+
+Evaludates `authState` (a unique new object) based on tokens in `tokenManager` or custom `isAuthenticated` callback, then emit `authStateChange` event with latest evaludated `authState`. By default, the evaluation process is driven by tokens change. Might be need to be triggered manually for users that pass override of `isAuthenticated` to the [configuration](#configuration-reference).
```

Comment:
Produces a unique `authState` object and emits an [authStateChange](#link-here) event. The [authState](#link-here) object contains tokens from the `tokenManager` and the results of the [isAuthenticated](#link-here) callback. By default, [isAuthenticated](#link-here) will be true if both `idToken` and `accessToken` are present. This logic can be customized by defining a custom [isAuthenticated](#link-here) function.